File: src/main/java/command_pattern/Waiter.java
package command_pattern;

import java.util.*;
import java.util.function.Consumer;

public class Waiter{

    private final List<Command> orders = new ArrayList<>();

    private final Map<String, Consumer<Command>> actionMap = new HashMap<String, Consumer<Command>>(){{
        put("class command_pattern.BakeChickenWingCommand", command ->
                System.out.println("No chicken wing, please ask something else!"));
        put("class command_pattern.BakeMuttonCommand", command -> {
            orders.add(command);
            System.out.printf("Order is added: %s @ %s \n", command.toString(), new Date());
        });
    }};

    public void setOrder(Command command) {
        actionMap.get(command.getClass().toString()).accept(command);
    }

    public void cancelOrder(Command command) {
        orders.remove(command);
        System.out.printf("Order is removed: %s @ %s \n", command.toString(), new Date());
    }

    public void notifyReceiver() {
        orders.forEach(Command::executeCommand);
    }

}

File: src/main/javascript/command_pattern/waiter.js
function Waiter() {

    let orders = [];

    const actionMap = {
        BakeChickenWingCommand(command) {
            console.warn(`No chicken wing, please ask something else!`);
        },
        BakeMuttonCommand(command) {
            orders.push(command);
            console.info(`Order is added: ${command.constructor.name}:${command.id} @ ${new Date()}`);
        },
        BakeCornCommand(command) {
            orders.push(command);
            console.info(`Order is added: ${command.constructor.name}:${command.id} @ ${new Date()}`);
        },
        default(command){
            console.info(`Command ${command.constructor.name} strategy not found`);
        }
    };

    function setOrder(command) {
        const fn = actionMap[command.constructor.name];
        typeof fn === 'function' ? fn(command) : actionMap['default'](command);
    }

    function cancelOrder(command) {
        orders = orders.filter(order => order.id === command.id);
        console.info(`Order is cancelled: ${command.constructor.name}:${command.id}`);
    }

    function notifyReceiver() {
        orders.forEach(order => order.executeCommand());
    }

    return {
        setOrder,
        cancelOrder,
        notifyReceiver
    };

}

module.exports = Waiter;